### PR TITLE
Added MCDC Report feature

### PR DESCRIPTION
**Pull request by Den1552**
## Summary

Added an MC/DC report feature that allows the user to generate a report for a specific line. The report opens in a separate WebviewPanel, similar to the test generation report. Lines that are interactable are visually highlighted in the editor.

## Usage
1) Ensure that `Statement+MCDC` is selected under `Settings → VectorCAST Test Explorer → Build → Coverage Kind`. If this setting is not selected, the coverage gutter will display the standard coverage icons and will not be interactable.

2) To generate a report, `right-click` on the coverage icon and choose `Get MCDC Report`. The report will open in a separate WebviewPanel.

3) In` Settings → VectorCAST Test Explorer → MCDC Coverage Gutter Icons`, you can choose to display either an `arrow` or an `M` as the icon in the coverage gutter.

## TODOs
- [x] Write a server-side implementation for retrieving the MCDC lines for the coverage icons, if necessary
- [x] Refactoring

### CODE REVIEW

File: src/coverage.ts [hunk 1]
```diff
@@ -50,21 +88,44 @@ export function initializeCodeCoverageFeatures(
...
   covered: number[],
-  uncovered: number[]
+  uncovered: number[],
+  partiallyCovered: number[]
 ) {
   const lineCount = activeEditor.document.lineCount;
   let lineIndex;
   // these are lists of line numbers
 
   for (lineIndex = 0; lineIndex < lineCount; lineIndex++) {
```
Comment (Zbigor) [inline src/coverage.ts:106]: The conditions are a bit convoluted and you have some repetition and a bunch of comments. You can try something like the following, but do double-check that the logic still holds and that your tests are ok

```javascript
for (lineIndex = 0; lineIndex < lineCount; lineIndex++) {
    const lineNumber = lineIndex + 1;
    const isMCDCLine = currentActiveUnitMCDCLines.includes(lineNumber);

    if (partiallyCovered.includes(lineNumber)) {
        (isMCDCLine ? partiallyCoveredDecorationsWithMCDC : partiallyCoveredDecorations)
            .push(getRangeOption(lineIndex));
        continue;
    }

    if (covered.includes(lineNumber)) {
        (isMCDCLine ? coveredDecorationsWithMCDC : coveredDecorations)
            .push(getRangeOption(lineIndex));
        continue;
    }

    if (uncovered.includes(lineNumber)) {
        (isMCDCLine ? uncoveredDecorationsWithMCDC : uncoveredDecorations)
            .push(getRangeOption(lineIndex));
        continue;
    }
}

```

File: src/coverage.ts [hunk 2]
```diff
@@ -77,14 +138,27 @@ let coverageStatusBarObject: vscode.StatusBarItem;
 function resetGlobalDecorations() {
   uncoveredDecorations = [];
   coveredDecorations = [];
+  partiallyCoveredDecorations = [];
```
Comment (Zbigor) [inline src/coverage.ts:141]: try packing decoration types and decorations into 2 lists, then iterating and reseting each element of a list with a loop

File: src/coverage.ts [hunk 3]
```diff
@@ -152,6 +266,12 @@ function deactivateCoverage() {
   // delete all decorations
   if (uncoveredDecorationType) uncoveredDecorationType.dispose();
```
Comment (Zbigor) [inline src/coverage.ts:267]: try iterating over a list and calling the condition followed by dispose() to avoid repetition

**Review by Zbigor — CHANGES_REQUESTED** [review]
Went through all the changes, everything looks good. I haven't seen anything weird, the main comment is about the suggested refactor for clarity regarding all those nested if conditions.

File: src/coverage.ts (hunk 2 above)
Comment (Den1552) [reply src/coverage.ts]: Refactored in ca80e6c05de81bc3fda70e854b82c0cee7479e9a.

File: src/coverage.ts (hunk 1 above)
Comment (Den1552) [reply src/coverage.ts:105]: Looks better and works. Refactored in 74cbc4ab5cb5196f0b6cec7aff88c11823f60b12.

**Review by Zbigor — APPROVED** [review]

**Pull request merged**